File: fixtures/scale/dart/large/lib/verify/determinism.dart
import '../core/hash_function.dart';

bool stableDigest(HashFunction hash, String sample) =>
    hash.digest(sample) == hash.digest(sample);

bool distinctDigests(HashFunction hash, String left, String right) {
  if (left == right) return true;
  return hash.digest(left) != hash.digest(right);
}

File: fixtures/scale/kotlin/large/src/cryptobench/verify/Determinism.kt
package cryptobench.verify

import cryptobench.core.HashFunction

fun stableDigest(hash: HashFunction, sample: String): Boolean =
    hash.digest(sample) == hash.digest(sample)

fun distinctDigests(hash: HashFunction, left: String, right: String): Boolean {
    if (left == right) return true
    return hash.digest(left) != hash.digest(right)
}
